### PR TITLE
Changed constants name MIN_TEAM_LENGTH value to 2 and the error message

### PR DESCRIPTION
**Pull request by S4KH**
#### Summary
Reduced minimum length of team name from 4 to 2 characters

#### Ticket Link
https://github.com/mattermost/platform/issues/4146

#### Checklist
- [x] Includes text changes and localization/constants files updated
- [x] Includes changes to model/team.go, team_test.go

### CODE REVIEW

**Comment by jasonblais** [comment]
@S4KH Highly appreciated!

Thanks for completing the contribution license agreement, will queue the pull request for review.

If you're not a member, please also consider joining our [community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors, to ask questions from the Mattermost core team, and to find new opportunities to help.

**Comment by jasonblais** [comment]
Many thanks again @S4KH!

During testing (Windows/Chrome), I found the following:

1) When trying to create a team name with only one character, the error message shows 4 as the minimum length, instead of 2

image

I think it can be resolved by updating `create_team.display_name.charLength` and `create_team.team_url.charLength` in the [localization file here](https://github.com/mattermost/platform/blob/master/webapp/i18n/en.json#L1104).

2) After creating a team "AB", the team URL requires at least four characters.

image

Expected: Team URL can also have a minimum length of two

3) The `config.json` setting changes seem unnecessary for this PR.

@S4KH Can you help take a look at the above cases?

**Comment by S4KH** [comment]
Got it. Will do in a sec.

**Comment by S4KH** [comment]
Have no idea why build is failing. 
Build Log : 

    FAIL
    coverage: 33.7% of statements
    exit status 1
    FAIL	github.com/mattermost/platform/model	33.408s
    make: *** [test-server] Error 1
    [TemplateProject] FAILED performing builders from: 'mattermost-platform-template'

**Comment by jwilander** [comment]
@S4KH you need to update the unit test in `model/team_test.go` around line 98. It's likely checking to make sure team names are the correct length and needs to be updated to match your other changes

**Comment by jasonblais** [comment]
Many thanks @S4KH! Highly appreciated.

If you're not a member yet, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and find new opportunities to help.

----

+1, tested on Windows 10 Chrome. I think the changes to `config.json` aren't necessary, but will defer to devs.

File: config/config.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
         "MaxIdleConns": 10,
         "MaxOpenConns": 10,
         "Trace": false,
-        "AtRestEncryptKey": ""
```
Comment (crspeller) [inline config/config.json]: Can you clear the changes to the config.json please?

Comment (S4KH) [inline config/config.json]: did it. Don't understand  why build fails. Says authentication error

Comment (crspeller) [inline config/config.json]: Sorry that's on our end. Re-running the build.

File: config/config.json [hunk 2]
```diff
@@ -125,8 +125,9 @@
         "EmailBatchingInterval": 30
     },
     "RateLimitSettings": {
-        "EnableRateLimiter": true,
+        "Enable": false,
```
Comment (jwilander) [inline config/config.json]: Looks like you missed a couple changes to the config file, can you revert these as well?

**Comment by enahum** [comment]
@S4KH this PR has merge conflicts and to resolve them please revert the changes made to the localization files, the only one that should have any modifications is `en.json` the rest should be modified by the [Mattermost Translation Server](http://translate.mattermost.com).

Thanks

**Comment by jwilander** [comment]
Sorry @S4KH, just need one more rebase and I can merge this in

**Comment by S4KH** [comment]
@jwilander Thanks for the help mate. Rebase means? :sos:

**Comment by jwilander** [comment]
Take a look at the answer to this [Stack Overflow question](http://stackoverflow.com/questions/17182624/contributing-to-project-on-github-how-to-rebase-my-pull-request-on-top-of-mast). Should help you rebase to fix the conflicts :)

**Comment by jasonblais** [comment]
Thanks so much @S4KH for working on the updates, excited to have this pull request merged soon!

Looks like there's just one more merge conflict with `webapp/utils/constants.jsx`, can you help update?

We had a large PR that was merged a couple of days ago, sorry you've had to rebase the submission a couple of times now

**Pull request merged**